src/julia_gc.c

```diff
@@ -728,26 +728,23 @@ jl_datatype_t * GAP_DeclareBag(jl_sym_t *      name,
                                1, large > 0);
 }
 
-// internal wrapper for jl_boundp to deal with API change in Julia 1.12
-static int gap_jl_boundp(jl_module_t *m, jl_sym_t *var)
-{
-#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 12
-    return jl_boundp(m, var, 1);
-#else
-    return jl_boundp(m, var);
-#endif
-}
-
 // Initialize the integration with Julia's garbage collector; in particular,
-// create Julia types for use in our allocations. The types will be stored
-// in the given 'module', and the MPtr type will be a subtype of 'parent'.
-//
-// If 'module' is NULL then 'jl_main_module' is used.
-// If 'parent' is NULL then 'jl_any_type' is used.
+// create Julia types for use in our allocations.
+// If 'defined(USE_GAP_INSIDE_JULIA)' (ie this function gets called from julia):
+// This function assumes that the types have already been declared in the
+// Julia module 'module' (e.g., by GAP_DeclareGapObj and GAP_DeclareBag).
+// In particular, 'module' may not be NULL.
+// If '!defined(USE_GAP_INSIDE_JULIA)' (ie this function gets called from GAP):
+// The types will be storied in the given 'module', or 'jl_main_module' if
+// 'module' is NULL.
 void GAP_InitJuliaMemoryInterface(jl_module_t *   module,
-                                  jl_datatype_t * parent)
+                                  jl_datatype_t * /* unused */)
```

Comment:
```suggestion
                                  jl_datatype_t * parent /* unused */)
```
The compiler complains otherwise:
```
/tmp/GAPROOT/src/julia_gc.c:741:63: error: omitting the parameter name in a function definition is a C2x extension [-Werror,-Wc2x-extensions]
                                  jl_datatype_t * /* unused */)
```
(https://github.com/gap-system/gap/actions/runs/16976387346/job/48126296520?pr=6058#step:7:287)